src/commands/generate.ts

```diff
@@ -55,10 +57,41 @@ function generate(toolbox: GluegunToolbox) {
     return
   }
 
+  const isAppIconGenerator = generator === "app-icon"
+
   // we need a name for this component
   const name = parameters.second
   if (!name) {
-    return warning(`⚠️  Please specify a name for your ${generator}: ignite g ${generator} MyName`)
+    const exampleName = isAppIconGenerator ? "all|ios|android|expo" : "MyName"
+
+    warning(
+      isAppIconGenerator
+        ? `⚠️  Please specify which icons you would like to generate:`
+        : `⚠️  Please specify a name for your ${generator}:`,
+    )
+    p()
+    command(`ignite g ${generator} ${exampleName}`)
+    return
+  }
+
+  if (generator === "app-icon") {
+    const { isValid, messages } = await validateAppIconGenerator(name as any)
+
+    if (!isValid) {
+      messages.forEach((message) => warning(message))
+      return
+    }
+
+    const isSuccessful = await generateAppIcons(name as any)
+
+    if (isSuccessful) {
+      heading(`App icons generated!`)
+      p(
+        "Uninstall the application from your simulator/emulator and re-build your app to see the changes!",
+      )
+    }
+
+    return
```

Comment:
The app-icon generator is a bit diff compared to other generators in that it takes preset parameters and doesn't copy files into the ./app dir. For that reason, we can branch it here. 

Down the road, this can possibly change for other icon generators (e.g. notifications).

We could make it more generic later to be more like: `ignite g icon app|notification ios|android|expo`.